node_modules/col-activities/tests/test-daily.js

```diff
@@ -0,0 +1,788 @@
+                                  ActivitiesTestUtil.assertDailyNotifications(dbCourse, [users.nico.me, users.annesophie.me, users.simon.me], function(emails) {
+
+                                    return callback();
+                                  });
+                                });
+                              });
+                            });
+                          });
+                        });
+                      });
+                    });
+                  });
+                });
+              });
+            });
+          });
+        });
+      });
+    });
+  });
+
+  /**
+   * Test that verifies that the email subject depends on which activities took place
+   */
+  it('changes the email subject depending on which activities took place', function(callback) {
+    setupCourseWithActivity(function(dbCourse, course, users) {
+
+      // Collect the daily notifications
+      var expectedEmailedUsers = [users.nico.me, users.annesophie.me, users.simon.me, users.paul.me];
+      ActivitiesTestUtil.assertDailyNotifications(dbCourse, expectedEmailedUsers, function(emails) {
+        var nicosEmail = getEmail(emails, users.nico.me.canvas_email);
+        var anneSophiesEmail = getEmail(emails, users.annesophie.me.canvas_email);
+        var simonsEmail = getEmail(emails, users.simon.me.canvas_email);
+        var paulsEmail = getEmail(emails, users.paul.me.canvas_email);
+
+        assert.ok(nicosEmail.email.subject.indexOf('commented on your asset ‘' + users.nico.asset.title + '’') > -1);
+        assert.strictEqual(anneSophiesEmail.email.subject, users.simon.me.canvas_full_name + ' replied to your comment ‘This is a comment by Anne-Sophie’');
+        assert.strictEqual(simonsEmail.email.subject, users.paul.me.canvas_full_name + ' commented on your whiteboard ‘' + users.simon.whiteboard.title + '’');
+        assert.strictEqual(paulsEmail.email.subject, 'Today\'s activity');
+
+        return callback();
+      });
+    });
+  });
+
+  /**
+   * Test that verifies that replies on an asset owner's comment are grouped as replies
+   */
+  it('groups replies on an asset owner\'s comment correctly', function(callback) {
+    setupCourse(function(dbCourse, course, users) {
+
+      // Simon creates a comment on his own comment which Nico then comments on
+      AssetsTestUtil.assertCreateComment(users.simon.client, course, users.simon.asset.id, 'This is a comment', null, function(topComment) {
+        AssetsTestUtil.assertCreateComment(users.nico.client, course, users.simon.asset.id, 'This is a reply', topComment.id, function(reply) {
+
+          // Simon should get an email notifying him of a reply
+          ActivitiesTestUtil.assertDailyNotifications(dbCourse, [users.simon.me], function(emails) {
+            var simonsEmail = getEmail(emails, users.simon.me.canvas_email);
+            assert.strictEqual(simonsEmail.email.subject, users.nico.me.canvas_full_name + ' replied to your comment ‘This is a comment’');
```

Comment:
Is there a way to verify that replies on the same comment are grouped together under that comment ?